src/plugins/intel_cpu/src/nodes/matmul.cpp

```diff
@@ -588,8 +684,21 @@ void MatMul::prepareParams() {
         dnnlBiasMemDesc = biasMemory->GetDescWithType<DnnlMemoryDesc>();
     }
 
-    MatMulKey key = {src0TransposedDesc, src1TransposedDesc, dnnlBiasMemDesc,
-                     dstDnnlDesc, *attr, selected_pd->getImplementationType()};
+    memory::desc descIn0 = src0TransposedDesc->getDnnlDesc();
+    memory::desc descIn1 = src1TransposedDesc->getDnnlDesc();
+    memory::desc descBias = dnnlBiasMemDesc ? dnnlBiasMemDesc->getDnnlDesc() : dnnl::memory::desc();
+    memory::desc descOut = dstDnnlDesc->getDnnlDesc();
+
+    SimplifyAs2D(descIn0, descIn1, descBias, descOut);
+
```

Comment:
This routine will inevitable introduce overheads to `prepareParams` call, but as far as I understand it, it is necessary only  for 6D case, so can we extract this logic to a some routine that can be initialized on the compile stage (since the rank is a constant value) and used polymorphically. The routine (m.b. some functor class) will be initialized on the compile stage since the rank is a constant value. These conversions:
```cpp
{DnnlExtensionUtils::makeDescriptor(descIn0),
                     DnnlExtensionUtils::makeDescriptor(descIn1),
                     dnnlBiasMemDesc ? DnnlExtensionUtils::makeDescriptor(descBias) : nullptr,
                     DnnlExtensionUtils::makeDescriptor(descOut),
                     *attr,
                     selected_pd->getImplementationType()
```
Should also be a part of the routine. So something like that:

```cpp
   std::function<key(...)> init_key;
   
   /*on the prepareParams*/
   if (rank==6) {
     init_key = [](...) {
       SimplifyAs2D //and other code we need
       return MatMulKey{...}
     } else {
        init_key = [](...) {
           //the old init code
           return MatMulKey{...} }
         }
```